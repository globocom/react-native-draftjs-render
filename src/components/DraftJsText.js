/*
 * Copyright (c) 2017, Globo.com (https://github.com/globocom)
 *
 * License: MIT
 */

// @flow

import React from 'react';
import { Text } from 'react-native';

import loadAttributes from '../loadAttributes';

import defaultStyles from './defaultStyles';
import type { DraftJsTextPropsType } from './types';

const DraftJsText = (props: DraftJsTextPropsType): any => {
  const { text, renderEmptyBlocks } = props;
  let textElements = text;

  if (textElements || renderEmptyBlocks) {
    textElements = loadAttributes({
      text: props.text,
      customStyles: props.customStyles,
      inlineStyles: props.inlineStyles,
      entityRanges: props.entityRanges,
      entityMap: props.entityMap,
      navigate: props.navigate,
      textProps: props.textProps,
      type: props.type,
    });

    const customStyle = props.customStyles ? props.customStyles[props.type] : undefined;
    const textAlignStyle = { textAlign: props.data.textAlignment };

    return (
      <Text
        style={[defaultStyles[props.type], textAlignStyle, customStyle]}
        {...props.textProps}
      >
        {textElements}
      </Text>
    );
  }
  return null;
};

DraftJsText.defaultProps = {
  text: '',
  data: {},
  inlineStyles: [],
  navigate: undefined,
};

export default DraftJsText;
