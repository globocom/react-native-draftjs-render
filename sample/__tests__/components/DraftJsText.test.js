/*
 * Copyright (c) 2017, Globo.com (https://github.com/globocom)
 *
 * License: MIT
 */

/* eslint-env jest */

import 'react-native';

import React from 'react';
import renderer from 'react-test-renderer';

import DraftJsText from '../../../src/components/DraftJsText';

it('renders correctly with a text', () => {
  const text = 'Hello World';
  const tree = renderer.create(<DraftJsText
    type="paragraph"
    text={text}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders null without a text', () => {
  const tree = renderer.create(<DraftJsText />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('extends a style with a customStyle', () => {
  const text = 'Hello World';
  const customStyles = {
    paragraph: {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
    },
  };
  const tree = renderer.create(<DraftJsText
    type="paragraph"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('extends a style with a customStyle from another type', () => {
  const text = 'Hello World';
  const customStyles = {
    blockquote: {
      fontSize: 18,
      fontWeight: 'normal',
      letterSpacing: -0.75,
      lineHeight: 32,
    },
  };
  const tree = renderer.create(<DraftJsText
    type="paragraph"
    text={text}
    customStyles={customStyles}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    navigate={() => null}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders text-align: left', () => {
  const text = 'Hello World';
  const data = { textAlignment: 'left' };
  const tree = renderer.create(<DraftJsText
    type="paragraph"
    text={text}
    data={data}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders text-align: right', () => {
  const text = 'Hello World';
  const data = { textAlignment: 'right' };
  const tree = renderer.create(<DraftJsText
    type="paragraph"
    text={text}
    data={data}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders text-align: center', () => {
  const text = 'Hello World';
  const data = { textAlignment: 'center' };
  const tree = renderer.create(<DraftJsText
    type="paragraph"
    text={text}
    data={data}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders empty blocks', () => {
  const text = '';
  const tree = renderer.create(<DraftJsText
    type="paragraph"
    text={text}
    inlineStyles={[]}
    entityRanges={[]}
    entityMap={{}}
    renderEmptyBlocks
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
