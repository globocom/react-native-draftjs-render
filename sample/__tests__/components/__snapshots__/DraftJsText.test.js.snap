// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extends a style with a customStyle 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 14,
        "fontWeight": "normal",
      },
      Object {
        "fontSize": 18,
        "fontWeight": "normal",
        "letterSpacing": -0.75,
        "lineHeight": 32,
      },
    ]
  }
>
  Hello World
</Text>
`;

exports[`extends a style with a customStyle from another type 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 14,
        "fontWeight": "normal",
      },
      undefined,
    ]
  }
>
  Hello World
</Text>
`;

exports[`renders correctly with a text 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontSize": 14,
        "fontWeight": "normal",
      },
      undefined,
    ]
  }
>
  Hello World
</Text>
`;

exports[`renders null without a text 1`] = `null`;
